src/Cart.js

```diff
@@ -0,0 +1,106 @@
+import firebaseConfig from './firebase';
+import { getDatabase, ref, onValue, remove } from 'firebase/database'
+import { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+
+const Cart = () => {
+    const [cart, setCart] = useState([]);
+
+    useEffect(() => {
+        // create a variable that will hold onto our database values
+        const database = getDatabase(firebaseConfig);
+        // create a variable that will reference our database
+        const cartRef = ref(database, `cart`);
+        onValue(cartRef, (response) => {
+            const data = response.val()
+            console.log(data)
+            const newArray = [];
+            for (let item in data) (
+                newArray.push({key: item, value: data[item]})
+            )
+            console.log(newArray)
+
+            const reduce = new Map(newArray.map(newItem => [newItem.value.description, { ...newItem, quantity: 0}]));
+            for (const {value} of newArray) reduce.get(value.description).quantity++;
+            const cartArray = Array.from(reduce.values());
+            console.log(cartArray)
+            setCart(cartArray)
```

Comment:
I love how you simplified the code, maybe you can remove console.log for tomorrow review